sktime/transformations/series/detrend/mstl.py

```diff
@@ -0,0 +1,248 @@
+        DataFrame, attempts to determine from endog. If endog is a ndarray, periods
+        must be provided.
+    windows : {int, array_like, None}, optional
+        Length of the seasonal smoothers for each corresponding period. Must be an odd
+        integer, and should normally be >= 7 (default). If None then default values
+        determined using 7 + 4 * np.arange(1, n + 1, 1) where n is number of seasonal
+        components.
+    lmbda : {float, str, None}, optional
+        The lambda parameter for the Box-Cox transform to be applied to endog prior to
+        decomposition. If None, no transform is applied. If `auto`, a value will be
+        estimated that maximizes the log-likelihood function.
+    iterate : int, optional
+        Number of iterations to use to refine the seasonal component.
+    stl_kwargs : dict, optional
+        Arguments to pass to STL.
+    return_components : bool, default=False
+        if False, will return only the MSTL transformed series
+        if True, will return the transformed series, as well as three components
+            as variables in the returned multivariate series (DataFrame cols)
+            "transformed" - the transformed series
+            "seasonal" - the seasonal component(s)
+            "trend" - the trend component
+            "resid" - the residuals after de-trending, de-seasonalizing
+
+    Attributes
+    ----------
+    trend_ : pd.Series
+        Trend component of series seen in fit.
+    seasonal_ : pd.Series or list of pd.Series
+        If `periods` is a single value, this contains the seasonal component of the
+        series observed during fitting.
+        If `periods` is a list of values, this can contain multiple pd.Series, each
+        corresponding to a different period.
+    resid_ : pd.Series
+        Residuals component of series seen in fit.
+
+    References
+    ----------
+    [1] https://www.statsmodels.org/dev/generated/statsmodels.tsa.seasonal.MSTL.html
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt  # doctest: +SKIP
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.transformations.series.detrend.mstl import MSTL
+    >>> y = load_airline()
+    >>> y.index = y.index.to_timestamp()
+    >>> mstl = MSTL(return_components=True)  # doctest: +SKIP
+    >>> fitted = mstl.fit(y)  # doctest: +SKIP
+    >>> res = fitted.transform(y)  # doctest: +SKIP
+    >>> res.plot()  # doctest: +SKIP
+    >>> plt.tight_layout()  # doctest: +SKIP
+    >>> plt.show()  # doctest: +SKIP
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "X_inner_mtype": "pd.Series",
+        "y_inner_mtype": "pd.Series",
+        "transform-returns-same-time-index": True,
+        "univariate-only": True,
+        "fit_is_empty": False,
+        "python_dependencies": "statsmodels",
```

Comment:
since you decided to implement the `_inverse_transform`, you should add the tags or it will always be skipped.

here is the relevant excerpt:

```
        # capability:inverse_transform = is inverse_transform implemented?
        "capability:inverse_transform": False,
        # valid values: boolean True (yes), False (no)
        # if True, _inverse_transform must be implemented
        # if False, exception is raised if inverse_transform is called,
        #   unless the skip-inverse-transform tag is set to True
```